src/chat/flagsReducers.js

```diff
@@ -91,11 +91,9 @@ const processFlagsForMessages = (state: FlagsState, messages: Message[]): FlagsS
 
 const rehydrate = (state: FlagsState, action: RehydrateAction): FlagsState => {
   const messagesState: { [narrow: string]: Message[] } =
-    action.payload && action.payload.accounts ? action.payload.narrows : {};
+    action.payload && action.payload.accounts ? action.payload.messages : {};
   // $FlowFixMe: Object.values
-  const arrayOfMessageArrays: Array<Message[]> = Object.values(messagesState);
-  const flattenedMessages: Message[] = Array.prototype.concat(...arrayOfMessageArrays);
-  return processFlagsForMessages(state, flattenedMessages);
+  return processFlagsForMessages(state, Object.values(messagesState));
```

Comment:
As the commit message on 9df067b23 which fixed the bug here says, let's make a practice of keeping `$FlowFixMe` as tightly focused as possible -- applying it to more than it has to is what caused this bug.

Here, that means applying to just the `Object.values` call. Could do that with a local variable, as in the old code; or I think it should also work to do something like
```
  return processFlagsForMessages(state,
    // $FlowFixMe: Object.values
    Object.values(messagesState));
```